Timestamp: 12/07/11 08:06:42 (10 months ago)
Author: Simon Marlow <marlowsd@…>
Children: 8a5f2fba4cc2b4cc6e07f1d0f7e351c6fb9965c4, 606f6e1cfcb2e79abaadcc5ed643817d2a4585d8, 72bfc815461354e806e5ef4a0a36db480aada28d, 3640d374f3501c0addb1be273ecc0f10f93be10d
Parents: eea40328004e3cad1fdd31004337e10e6ae5fc52
git-author: Simon Marlow <marlowsd@…> (12/07/11 07:46:27)
git-committer: Simon Marlow <marlowsd@…> (12/07/11 08:06:42)
Message: Define getNumberOfProcessors() even when !THREADED_RTS
Index: includes/rts/OSThreads.h
===================================================================
--- includes/rts/OSThreads.h	(revision 67a3085716af0e2334dece594ec42fcf37d43bda)
+++ includes/rts/OSThreads.h	(revision 8b48562e8cd773b4bc162edb4fc6d05a7654c7e1)
@@ -184,6 +184,5 @@
 extern rtsBool broadcastCondition ( Condition* pCond );
 extern rtsBool signalCondition    ( Condition* pCond );
-extern rtsBool waitCondition      ( Condition* pCond, 
-				    Mutex* pMut );
+extern rtsBool waitCondition      ( Condition* pCond, Mutex* pMut );
 
 //
@@ -202,5 +201,4 @@
 
 // Processors and affinity
-nat  getNumberOfProcessors (void);
 void setThreadAffinity     (nat n, nat m);
 #endif // !CMINUSMINUS
@@ -214,10 +212,15 @@
 #endif /* defined(THREADED_RTS) */
 
+#ifndef CMINUSMINUS
 //
 // Support for forkOS (defined regardless of THREADED_RTS, but does
 // nothing when !THREADED_RTS).
 //
-#ifndef CMINUSMINUS
 int forkOS_createThread ( HsStablePtr entry );
+
+//
+// Returns the number of processor cores in the machine
+//
+nat getNumberOfProcessors (void);
 #endif
 
Index: rts/Linker.c
===================================================================
--- rts/Linker.c	(revision 021a0dd265ff34c1e292813c06185eff1d6b5c1c)
+++ rts/Linker.c	(revision 8b48562e8cd773b4bc162edb4fc6d05a7654c7e1)
@@ -851,4 +851,5 @@
       SymI_HasProto(resumeThread)                       \
       SymI_HasProto(setNumCapabilities)                 \
+      SymI_HasProto(getNumberOfProcessors)              \
       SymI_HasProto(resolveObjs)                        \
       SymI_HasProto(stg_retryzh)                        \
Index: rts/posix/OSThreads.c
===================================================================
--- rts/posix/OSThreads.c	(revision 8b75acd3ca25165536f18976c8d80cb62ad613e4)
+++ rts/posix/OSThreads.c	(revision 8b48562e8cd773b4bc162edb4fc6d05a7654c7e1)
@@ -309,3 +309,8 @@
 }
 
-#endif /* !defined(THREADED_RTS) */
+nat getNumberOfProcessors (void)
+{
+    return 1;
+}
+
+#endif
Index: rts/win32/OSThreads.c
===================================================================
--- rts/win32/OSThreads.c	(revision dad0d225ec5117b715efc0c44c4fa0d606660381)
+++ rts/win32/OSThreads.c	(revision 8b48562e8cd773b4bc162edb4fc6d05a7654c7e1)
@@ -309,3 +309,8 @@
 }
 
+nat getNumberOfProcessors (void)
+{
+    return 1;
+}
+
 #endif /* !defined(THREADED_RTS) */
